srtcore/group.cpp

```diff
@@ -780,7 +780,12 @@ void CUDTGroup::getOpt(SRT_SOCKOPT optname, void* pw_optval, int& w_optlen)
     return ps->core().getOpt(optname, (pw_optval), (w_optlen));
 }
 
+#if __cplusplus >= 202002L
+// std::unary_function is deprecated in C++11, removed in C++17.
+struct HaveState
+#else
 struct HaveState : public unary_function<pair<SRTSOCKET, SRT_SOCKSTATUS>, bool>
+#endif
```

Comment:
You might check if this unary_function is needed at all. If the code compiles just as well without it in C++98 mode, it can be simply removed.

Reply:
The `HaveState` struct is being removed in #2442.